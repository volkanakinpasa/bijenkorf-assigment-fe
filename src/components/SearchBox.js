import '../styles/searchBox.scss';

import Button from './Button';
import SearchBoxIcon from './SearchBoxIcon';

function SearchBox(props) {
  const { onSearchInputChange } = props;

  return (
    <div className="input-container relative">
      <input
        name="search"
        type="search"
        className="search"
        placeholder="Zooken"
        autoComplete="off"
        onChange={onSearchInputChange}
        data-testid="search"
      />

      <Button
        label="Zooken"
        icon={<SearchBoxIcon />}
        type="submit"
        onClick={(e) => {
          e.preventDefault();
          console.log(e);
        }}
      ></Button>
    </div>
  );
}

export default SearchBox;
