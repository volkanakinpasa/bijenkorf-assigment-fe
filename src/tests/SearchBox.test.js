import { fireEvent, render } from '@testing-library/react';

import SearchBox from '../components/SearchBox';

describe('SearchBox tests', () => {
  it('creates a snapshot', () => {
    const { asFragment } = render(<SearchBox />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders empty input', () => {
    const { getByTestId } = render(<SearchBox />);

    const input = getByTestId('search');

    expect(input.value).toBe('');
  });

  it('should get search value when input is changed', () => {
    const { getByTestId } = render(<SearchBox />);
    const input = getByTestId('search');

    fireEvent.change(input, {
      target: { value: 'lorem' },
    });

    expect(input.value).toBe('lorem');
  });
});
