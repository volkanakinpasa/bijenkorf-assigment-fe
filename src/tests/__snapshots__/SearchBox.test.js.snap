// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox tests creates a snapshot 1`] = `
<DocumentFragment>
  <div
    class="input-container relative"
  >
    <input
      autocomplete="off"
      class="search"
      data-testid="search"
      name="search"
      placeholder="Zooken"
      type="search"
    />
    <button
      aria-label="Zooken"
      type="submit"
    >
      <img
        src="search-icon.svg"
      />
    </button>
  </div>
</DocumentFragment>
`;
